payjoin/src/receive/v1/mod.rs

```diff
@@ -1070,4 +1088,128 @@ pub(crate) mod test {
+                x_only,
+                (
+                    vec![TapLeafHash::from_script(&ScriptBuf::new(), LeafVersion::TapScript)],
+                    dummy_key_source.clone(),
+                ),
+            );
+            input.tap_internal_key = Some(x_only);
+        }
+
+        for output in &mut processed_psbt.outputs {
+            output.redeem_script = Some(ScriptBuf::new());
+            output.witness_script = Some(ScriptBuf::new());
+            let (key, value) = dummy_bip32_derivation();
+            output.bip32_derivation.insert(key, value);
+            output.tap_key_origins.insert(
+                x_only,
+                (
+                    vec![TapLeafHash::from_script(&ScriptBuf::new(), LeafVersion::TapScript)],
+                    dummy_key_source.clone(),
+                ),
+            );
+            output.tap_internal_key = Some(x_only);
+        }
+
+        let provisional = provisional_proposal_from_test_vector(proposal);
+        let payjoin_proposal = provisional.prepare_psbt(processed_psbt);
+
+        for input in &payjoin_proposal.payjoin_psbt.inputs {
+            assert!(input.partial_sigs.is_empty());
+            assert!(input.redeem_script.is_none());
+            assert!(input.witness_script.is_none());
+            assert!(input.sha256_preimages.is_empty());
+            assert!(input.hash160_preimages.is_empty());
+            assert!(input.hash256_preimages.is_empty());
+            assert!(input.ripemd160_preimages.is_empty());
+            assert!(input.tap_scripts.is_empty());
+            assert!(input.proprietary.is_empty());
+            assert!(input.unknown.is_empty());
+            assert!(input.bip32_derivation.is_empty());
+            assert!(input.tap_key_origins.is_empty());
+            assert!(input.tap_internal_key.is_none());
+        }
+
+        for output in &payjoin_proposal.payjoin_psbt.outputs {
+            assert!(output.redeem_script.is_none());
+            assert!(output.witness_script.is_none());
+            assert!(output.bip32_derivation.is_empty());
+            assert!(output.tap_key_origins.is_empty());
+            assert!(output.tap_internal_key.is_none());
+        }
+    }
+
+    fn dummy_bip32_derivation() -> (PublicKey, (Fingerprint, DerivationPath)) {
+        let fingerprint = bitcoin::bip32::Fingerprint::from([0x12, 0x34, 0x56, 0x78]);
+        let derivation_path = bitcoin::bip32::DerivationPath::from(vec![
+            ChildNumber::from_hardened_idx(0).expect("index in range"),
+            ChildNumber::from_normal_idx(1).expect("index in range"),
+            ChildNumber::from_hardened_idx(2).expect("index in range"),
+            ChildNumber::from_normal_idx(3).expect("index in range"),
+        ]);
+
+        let (public_key, _) = dummy_key_pair();
+        (public_key, (fingerprint, derivation_path))
+    }
+
+    fn dummy_key_pair() -> (PublicKey, SecretKey) {
+        let secp = Secp256k1::new();
```

Comment:
Why not just use the global context?

Reply:
because we don't enable `global-context` feature by default. Should we?